src/plugins/kibana/server/routes/api/ingest/register_bulk.js

```diff
@@ -0,0 +1,98 @@
+import { Promise } from 'bluebird';
+import { parse, transform } from 'csv';
+import _ from 'lodash';
+import hi from 'highland';
+import { patternToIngest } from '../../../../common/lib/convert_pattern_and_ingest_name';
+import { PassThrough } from 'stream';
+import bulkRequestSchema from '../../../lib/schemas/bulk_request_schema';
+
+export function registerBulk(server) {
+  server.route({
+    path: '/api/kibana/{id}/_bulk',
+    method: 'POST',
+    config: {
+      payload: {
+        output: 'stream',
+        maxBytes: 1024 * 1024 * 1024
+      },
+      validate: {
+        payload: bulkRequestSchema
+      }
+    },
+    handler: function (req, reply) {
+      const boundCallWithRequest = _.partial(server.plugins.elasticsearch.callWithRequest, req);
+      const indexPattern = req.params.id;
+      const usePipeline = req.payload.pipeline;
+      const csv = req.payload.csv;
+      const fileName = csv.hapi.filename;
+      const responseStream = new PassThrough();
+      const parser = parse({
+        columns: true,
+        auto_parse: true,
+        delimiter: _.get(req.payload, 'delimiter', ',')
+      });
+
+      let currentLine = 2; // Starts at 2 since we parse the header separately
+
+      responseStream.write('[');
+      csv.pipe(parser);
+
+      hi(parser)
```

Comment:
The definition of MVP is too loose to argue over. I just meant "it's not perfect, but it gets the job done".

> it's generally hard to make the case that code is quality if it can't be unit tested

I disagree, because I wrote some pretty extensive functional tests that cover everything the unit tests would, and more.

> It also makes it impossible to unit test

I also disagree with that. I could pull the non-trivial anonymous functions out into their own modules and write unit tests for those. I didn't because I felt like they'd be re-testing what I already covered in the functional tests.

The more I think about it, I don't think what you're asking for is that much work. The only reason I'm opposed to doing it in this PR is that my current goal isn't to create a general purpose library. I've intentionally bound this code to csv parsing because I haven't thought through all the ins-and-outs of exposing this as a general purpose library, which was unnecessary to support the csv upload UI. I'd rather wait to do that until I tackle json file uploads in the UI, which will give me another concrete use case to guide the design of the library. That's just generally how I like to approach things to limit my natural tendency to over engineer. Build the naive solution first, refactor as necessary when requirements change.

It seems like testing is your biggest concern. Like I said I can pull the anonymous functions out and add unit tests for them. Would that be a good compromise?